src/contracts/validateDelegation.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { Provider } from '@wagmi/core';
+import { ethers } from 'ethers';
+import { toast } from 'react-toastify';
+import TransactionToast from '../components/TransactionToast';
+import { createMultiStepsTransactionToast, showErrorTransactionToast } from '../utils/toast';
+
+export const validateDelegation = async (
+  user: string,
+  DelegateAddress: string,
+  provider: Provider,
+  validateState: boolean,
+  contract: ethers.Contract,
+): Promise<void> => {
+  try {
+    let tx: ethers.providers.TransactionResponse;
+    if (validateState === true) {
+      tx = await contract.addDelegate(user, DelegateAddress);
+    } else {
+      tx = await contract.removeDelegate(user, DelegateAddress);
+    }
+
+    await createMultiStepsTransactionToast(
+      {
+        pending: 'Submitting the delegation...',
+        success: 'Congrats! the delegation is active',
```

Comment:
message need to change if it's a add or remove. Or just a generate Delegatation has been well updated